### DISCUSSION TITLE
What maths is used to determine the signed distance from a mesh?

### DISCUSSION THREAD

**Discussion by ljmartin** [libigl/libigl#1735] [Q&A] [answered]
Hi libigl,
Ive really enjoyed using this library, so thanks! The tutorials are great too.

Can anyone please tell me what maths / algorithm is used to calculate the signed distance from a point to a mesh?

It's quite fast for the mesh I'm using, but I want to see if I can write it in JAX (python) to run on a GPU and get it even faster.

**Comment by jdumas** [comment, marked as answer]
The unsigned distance is computed using a AABB tree for acceleration, while the sign of the distance (i.e. inside/outside) can be determined using either:
1. The normal of the closest face.
2. The [generalized winding number](https://www.dgp.toronto.edu/projects/fast-winding-numbers/), which offers a more robust solution for imperfect data.

**Reply by ljmartin** [reply]
thnx!
